UnitsNet.Tests/CustomCode/AmountOfSubstanceTests.cs

```diff
@@ -122,5 +122,65 @@ public void VolumeSolutionFromComponentMassAndDesiredConcentration(
 
             AssertEx.EqualTolerance(expectedSolutionVolumeValue, volumeSolution.As(expectedSolutionVolumeUnit), tolerence);
         }
+
+        [Fact]
+        public void From_Mass_And_MolarMass_ReturnsTheMassDividedByTheMolarMass()
+        {
+            var mass = Mass.From(10, MassUnit.Gram);
+            var molarMass = MolarMass.From(2, MolarMassUnit.GramPerMole);
+            var expectedAmountOfSubstance = AmountOfSubstance.From(5, AmountOfSubstanceUnit.Mole);
+
+            var amountOfSubstance = AmountOfSubstance.FromMass(mass, molarMass);
+
+            Assert.Equal(expectedAmountOfSubstance, amountOfSubstance);
+        }
+
+        [Theory]
+        [InlineData(5, AmountOfSubstanceUnit.Mole,
+            10, MolarEnergyUnit.JoulePerMole,
+            50, EnergyUnit.Joule)]
+        public void Multiplying_AmountOfSubstance_By_MolarEnergy_ReturnsEnergy(double amountOfSubstanceValue, AmountOfSubstanceUnit amountOfSubstanceUnit,
```

Comment:
If you only have one `InlineData`, I think a regular `[Fact]` is better and just inline the values in the code instead of passing as parameters. It was not particularly readable as it stands.

Theory is mostly useful when passing multiple variations of parameters.
Same for other places.

Reply:
Basically do what you did in AreaTests, it's short and sweet.

Reply:
I was mostly lazy (_and partly afraid that picking another value would turn up a rounding error_)- but technically, for a "complete" set of values we need to include `NaN` / `Infinity` as well as both base and non-base units...

Reply:
I've updated the tests (removing the [Theory])